assets/app/scripts/controllers/image.js

```diff
@@ -56,6 +60,51 @@ angular.module('openshiftConsole')
       );
     });
 
+    function updateTagsByName(){
+      $scope.tagsByName = {};
+      angular.forEach($scope.imageStream.spec.tags, function(tag){
+        $scope.tagsByName[tag.name] = $scope.tagsByName[tag.name] || {};
+        $scope.tagsByName[tag.name].name = tag.name;
+        $scope.tagsByName[tag.name].spec = angular.copy(tag);
+
+        // Split the name into useful parts, recalculate name to a common form to
+        // include self-references within the same IS
+        var from = $scope.tagsByName[tag.name].spec.from;
+        if (from) {
+          var splitChar = "";
+          if (from.kind === "ImageStreamImage") {
+            splitChar = "@";
+          }
+          else if (from.kind === "ImageStreamTag") {
+            splitChar = ":";
+          }
+          from._nameConnector = splitChar || null;
+          var parts = from.name.split(splitChar);
```

Comment:
passing a limit to split doesnt do what you would expect, it returns up to the limit elements in the array and then just dumps the rest, instead of only splitting limit times.  I can check if the length is longer than 2 and join anything else that got split.